elasticdl/python/worker/task_data_service.py

```diff
@@ -159,11 +188,9 @@ def _gen(self):
                     self._pending_save_model_task = task
                     continue
 
-                self._record_count += task.end - task.start
-                self._pending_tasks_with_counts.append(
-                    (task, self._record_count)
-                )
-                if len(self._pending_tasks_with_counts) == 1:
+                self._cur_record_count = task.end - task.start
```

Comment:
When getting the second task, self._cur_record_count will be overwritten and results in error later.

Reply:
Tasks are handled sequentially?